corgi/tasks/prod_defs.py

```diff
@@ -67,6 +69,87 @@ def update_products() -> None:
                 parse_product_version(pd_product_version, product, product_node)
 
 
+def _find_by_cpe(cpe_patterns: list[str]) -> list[str]:
+    """Given a list of CPE patterns find all the CPEs from ET Variants which match it"""
+    if not cpe_patterns:
+        return []
+    regex_query = Q()
+    for pattern in cpe_patterns:
+        regex = pattern.replace(".", "\\.").replace("*", ".*")
```

Comment:
I guess based on the logic here, cpe_patterns are always glob expressions and not proper regular expressions. If that's ever not the case, the matching code here will either break or give surprising results.

I looked briefly at fnmatch, which can translate a glob to a regex for us and might be slightly more reliable. But I don't see any way to check if something really is a glob expression first.

I guess we could use `re.compile()` first, catch any exception if it's really a glob and not a valid regex, then call `fnmatch.translate()` instead. But that also seems error-prone since a glob expression like "some.*.pattern" might look like a regex, even though it's really a glob.

Feel free to implement whatever you think is best, I'm just nervous we have to rely on the prod-defs patterns always being correct without any validation.
```suggestion
        regex = fnmatch.translate(pattern)
```

Reply:
I took this straight from SDEngine PSModule.find_by_cpe. I use the values a little differently however. In SDEngine we're searching from an Errata CPE, matching back to modules and trying to find the 'best' match. In Corgi we are searching from PSModule, matching back to Errata CPEs. We want to find all matches.

We store all the matches on the Product Version (which matches to a ps_module in prod_defs), then later we try to match those to ProductStreams using their versions.